testing/unittests/jlibtests.cpp

```diff
@@ -617,4 +618,302 @@ CPPUNIT_TEST_SUITE_REGISTRATION( JlibQuantileTest );
 CPPUNIT_TEST_SUITE_NAMED_REGISTRATION( JlibQuantileTest, "JlibQuantileTest" );
 
 
+/* =========================================================== */
+
+class JlibReaderWriterTest : public CppUnit::TestFixture
+{
+    CPPUNIT_TEST_SUITE(JlibReaderWriterTest);
+    CPPUNIT_TEST(testCombinations);
+    CPPUNIT_TEST_SUITE_END();
+
+    const static unsigned spinScaling = 1000;
+
+    static unsigned spinCalculation(unsigned prev, unsigned scale)
+    {
+        unsigned value = prev;
+        for (unsigned i = 0; i < scale*spinScaling; i++)
+        {
+            value = (value * 0x1234FEDB + 0x87654321);
+        }
+        return value;
+    }
+
+    class Reader : public Thread
+    {
+    public:
+        Reader(IRowQueue & _source, Semaphore & _doneSem, unsigned _workScale)
+            : Thread("Reader"), source(_source), doneSem(_doneSem), workScale(_workScale), work(0)
+        {
+        }
+
+        virtual int run()
+        {
+            loop
+            {
+                const void * next;
+                if (!source.dequeue(next))
+                    break;
+                if (!next)
+                    break;
+                std::atomic<byte> * value = (std::atomic<byte> *)next;
+                (*value)++;
```

Comment:
I don't know of one - this would be cleaner if the origin allocation was an array of atomic bytes.